Iframe fullscreen button

jrchudy opened the pull request:
Opening this PR to discuss changes to HTML before updating failing test cases to this new HTML structure.

The HTML structure for iframes changed from:
```HTML
<figure class="embed-block -chaise-post-load" style="">
    <figcaption class="embed-caption" style="">Chaise</figcaption>
    <iframe src="https://dev.isrd.isi.edu/chaise/search" width="800" height="300" ></iframe>
</figure>
```

to when the caption is on top:
```HTML
<figure class="embed-block -chaise-post-load" style="">
    <div class="figcaption-wrapper" style="width: 800px;">
        <figcaption class="embed-caption" style="">Chaise</figcaption>
        <div class="iframe-btn-container pull-right"><a class="chaise-btn chaise-btn-secondary chaise-btn-iframe" href="https://dev.isrd.isi.edu/chaise/search" target="_blank">
            <span class="glyphicon glyphicon-fullscreen"></span> Full screen</a>
        </div>
    </div>
    <iframe src="https://dev.isrd.isi.edu/chaise/search" width="800" height="300" ></iframe>
</figure>
```

When caption is on the bottom, it looks like:
```HTML
<figure class="embed-block -chaise-post-load" style="">
    <div class="iframe-btn-container pull-right"><a class="chaise-btn chaise-btn-secondary chaise-btn-iframe" href="https://dev.isrd.isi.edu/chaise/search" target="_blank">
        <span class="glyphicon glyphicon-fullscreen"></span> Full screen</a>
    </div>
    <iframe src="https://dev.isrd.isi.edu/chaise/search" width="800" height="300" ></iframe>
    <figcaption class="embed-caption" style="">Chaise</figcaption>
</figure>
```

The other concern is that this would change all of our documentation examples for iframes too.

`fullscreen-off` class is added as a class in chaise that has to be attached to the figure so the selector reads like below (added in [PR in chaise repo](https://github.com/informatics-isi-edu/chaise/pull/2047)):
```css
figure.fullscreen-off .iframe-btn-container {
    display: none;
}
```

jrchudy commented on js/utils/helpers.js:
```diff
@@ -1924,8 +1924,12 @@
                                         }
                                         break;
                                     default:
+                                        if (attr[0] == "width") {
```
To accomplish getting the button to pull right to the same width as the iframe, I capture the width value for later use as part of the div that wraps the figcaption and fullscreen button (when caption is at top).

RFSH replied:
What you've done makes sense, Just make sure you've documented it somewhere. You should explain what is the expected behavior (where it should be aligned) and why you did it this way, so later we won't be confused by it.

RFSH commented on js/utils/helpers.js:
```diff
@@ -1969,11 +1973,20 @@
+                            var contentsWidth = 'style="width: ' + (frameWidth ? (frameWidth + (isNaN(parseInt(frameWidth)) ? "" : "px")) : "100%")+ ';"';
+
+                            // text align to pull button to the right
+                            var buttonHtml = '<div class="iframe-btn-container pull-right"><a class="chaise-btn chaise-btn-secondary chaise-btn-iframe" href="' + iframeSrc + '" target="_blank"><span class="glyphicon glyphicon-fullscreen"></span> Full screen</a></div>';
```
Since `pull-right` is a class specific too Bootstrap (and therefore Chaise environment) let's just remove it and make sure the `iframe-btn-container` rule in Chaise does the `float: right`

RFSH commented:
Apart from the small comment about the `pull-right`, I think the structure looks good. In general as long as what you have works in Chaise and is general enough, it should be good to go.

jrchudy commented:
Added/updated comment in `helpers.js` to communicate what the buttonHtml variable is for. Updated examples in documentation and further cleaned them up. Added a new example for the `fullscreen-off` class.
